packages/appsync-modelgen-plugin/src/utils/process-connections-v2.ts

```diff
@@ -10,12 +10,41 @@ import { processHasManyConnection } from './process-has-many';
 
 // TODO: This file holds several references to utility functions in the v1 process connections file, those functions need to go here before that file is removed
 
+function getBelongsToConnectedField(field: CodeGenField, model: CodeGenModel, connectedModel: CodeGenModel, connectionInfo: CodeGenDirective): CodeGenField | undefined {
+  if(connectionInfo.arguments.fields) {
+    let indexDirective = flattenFieldDirectives(model).find(dir => {
+      return dir.name === 'index' && dir.fieldName === connectionInfo.arguments.fields[0];
+    });
+
+    if(indexDirective) {
```

Comment:
Index directive is required for hasMany-belongsTo but I am not sure it is also used in hasOne-belongsTo relations.

Reply:
It isn't required, but I found it wasn't getting the correct connected field in that case, I added this because it wasn't making the correct link